src/serverenvironment.cpp

```diff
@@ -620,7 +620,8 @@ void ServerEnvironment::savePlayer(RemotePlayer *player)
 PlayerSAO *ServerEnvironment::loadPlayer(RemotePlayer *player, bool *new_player,
 	session_t peer_id, bool is_singleplayer)
 {
-	PlayerSAO *playersao = new PlayerSAO(this, player, peer_id, is_singleplayer);
+	auto playersao_u = std::make_unique<PlayerSAO>(this, player, peer_id, is_singleplayer);
+	auto playersao = playersao_u.get();
```

Comment:
I’d move this `get()` right above `addActiveObject` to keep the unique pointer unique for as long as feasible.

Reply:
The raw ptr is not meant to be owning, so the unique ptr is always uniquely owned.
Anyway, will do.